Foreman 3.18 release notes (#2275)
theforeman/theforeman.org: _includes/manuals/3.18/1.2_release_notes.md

### Headline features

#### SSH certificate support for remote execution

> Capsules can now use SSH certificates signed by a trusted certificate authority (CA) during remote execution connections. This feature avoids the default Trust on First Use (TOFU) model and provides centralized trust management and revocation capabilities.

Reviewer: ```suggestion
Smart Proxies can now use SSH certificates signed by a trusted certificate authority (CA) during remote execution connections. This feature avoids the default Trust on First Use (TOFU) model and provides centralized trust management and revocation capabilities.
```

* Deprecate ForemanModal in the core ([#39048](https://projects.theforeman.org/issues/39048))
* saltstack_setup still uses chkconfig ([#39043](https://projects.theforeman.org/issues/39043))
* pry-remote does not work at all when pry-byebug is enabled ([#39039](https://projects.theforeman.org/issues/39039))
* fix action button handling for plugins ([#39038](https://projects.theforeman.org/issues/39038))
* BIOS info is not populated in All hosts page and in Host Details tab ([#39036](https://projects.theforeman.org/issues/39036))
* Add new dynflow table to schema dumper ignore list ([#39035](https://projects.theforeman.org/issues/39035))
* Replace useForemanModal in HostsIndex ([#39027](https://projects.theforeman.org/issues/39027))
* Hide taxonomy options from API docs for taxonomy resources ([#39025](https://projects.theforeman.org/issues/39025))
* Hosts index page incorrectly uses array index as React key for table rows ([#39016](https://projects.theforeman.org/issues/39016))
* User.current not reloadable in dev console environment ([#39014](https://projects.theforeman.org/issues/39014))

> *  Scoped order is ignored, it's forced to be batch order error displayed on application startup ([#39007](https://projects.theforeman.org/issues/39007))

Reviewer 2: Extra space
```suggestion
* Scoped order is ignored, it's forced to be batch order error displayed on application startup ([#39007](https://projects.theforeman.org/issues/39007))
```

* Support for Agama installer (SLES16) ([#38877](https://projects.theforeman.org/issues/38877))
* Search is cleared when using bottom pagination on All Hosts page ([#38858](https://projects.theforeman.org/issues/38858))
* Move AutocompleteInput to foremanCore ([#38852](https://projects.theforeman.org/issues/38852))
* Update InputFactory component to PF5 ([#38842](https://projects.theforeman.org/issues/38842))
* Remove ACD plugin from installer ([#38841](https://projects.theforeman.org/issues/38841))
* Foreman Smart-Proxy 3.16 breaks ipmi support ([#38783](https://projects.theforeman.org/issues/38783))
* Introduce SSH cert support in templates ([#38499](https://projects.theforeman.org/issues/38499))
* Automatic firmware selection for VMware is not working on the \`compute_attributes\` form ([#37834](https://projects.theforeman.org/issues/37834))
* Add snapshot tests for global_registration.erb ([#37789](https://projects.theforeman.org/issues/37789))

#### Foreman - Compute resources
* Not all old GCE files are cleaned up ([#39020](https://projects.theforeman.org/issues/39020))

#### Foreman - Compute resources - OpenStack

> * unable to choose a fixed ip openstack neutron ([#38742](https://projects.theforeman.org/issues/38742))

Reviewer 2: Make capitalization consistent

```suggestion
* Unable to choose a fixed ip openstack neutron ([#38742](https://projects.theforeman.org/issues/38742))
```

* host_param macro & transient host parameters ([#38939](https://projects.theforeman.org/issues/38939))

#### Foreman - Plugins
* SELinux prevents foreman_kubevirt from connecting to the OpenShift API ([#38987](https://projects.theforeman.org/issues/38987))

#### Foreman - Reporting
* Add Cloud Billing Metadata inclusion options to Host - Installed Products report template ([#38920](https://projects.theforeman.org/issues/38920))

#### Foreman - Security
* CVE-2025-9572: GraphQL API permission bypass leads to information disclosure ([#38913](https://projects.theforeman.org/issues/38913))

#### Foreman - Templates
* Dualstack fallback is broken ([#39046](https://projects.theforeman.org/issues/39046))

#### Foreman - Tests
* Subnet factories use the wrong proxy for HTTPBoot feature ([#39045](https://projects.theforeman.org/issues/39045))
* Remove static wrapper methods for host fixtures in TemplateSnapshotService ([#39044](https://projects.theforeman.org/issues/39044))
* "should update libvirt compute resource with locs" test is flaky due to array ordering differences ([#39037](https://projects.theforeman.org/issues/39037))

> * preseed snapshots are not validated in tests ([#39012](https://projects.theforeman.org/issues/39012))

Reviewer 2: ```suggestion
* Preseed snapshots are not validated in tests ([#39012](https://projects.theforeman.org/issues/39012))
```

* cloud-init snapshots are not validated in tests ([#39006](https://projects.theforeman.org/issues/39006))

> * provisioning snapshots for Debian use RHEL-style partition table ([#39005](https://projects.theforeman.org/issues/39005))

Reviewer 2: ```suggestion
* Provisioning snapshots for Debian use RHEL-style partition table ([#39005](https://projects.theforeman.org/issues/39005))
```

* Update ForemanContext mock to include all metadata  ([#39004](https://projects.theforeman.org/issues/39004))
* Improve output of KS validation test ([#39001](https://projects.theforeman.org/issues/39001))
* Kickstart snapshots for EL10 are missing during tests ([#38999](https://projects.theforeman.org/issues/38999))

> * refactor Slot.test.js to react testing library ([#38971](https://projects.theforeman.org/issues/38971))

Reviewer 2: ```suggestion
* Refactor Slot.test.js to react testing library ([#38971](https://projects.theforeman.org/issues/38971))
```
